components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -1537,6 +1542,74 @@ public void makeAPIKeysForwardCompatible(API api) throws APIManagementException
         return failedEnvironment;
     }
 
+    /**
+     * This method returns the previous versions of a given API
+     * @param api
+     * @return oldPublishedAPIList
+     * @throws APIManagementException
+     */
+    private List<APIIdentifier> getOldPublishedAPIList(API api) throws APIManagementException {
+        List<APIIdentifier> oldPublishedAPIList = new ArrayList<APIIdentifier>();
+        List<API> apiList = getAPIsByProvider(api.getId().getProviderName());
+        APIVersionComparator versionComparator = new APIVersionComparator();
+        for (API oldAPI : apiList) {
+            if (oldAPI.getId().getApiName().equals(api.getId().getApiName()) &&
+                    versionComparator.compare(oldAPI, api) < 0 &&
+                    (oldAPI.getStatus().equals(APIStatus.PUBLISHED))) {
+                oldPublishedAPIList.add(oldAPI.getId());
+            }
+        }
+
+        return oldPublishedAPIList;
+    }
+
+    /**
+     * This method used to send notifications to the previous subscribers of older versions of a given API
+     * @param api
+     * @param apiIdentifiers
+     * @throws APIManagementException
+     */
+    private void sendEmailNotification(API api, List<APIIdentifier> apiIdentifiers) throws APIManagementException {
+        try {
+            String isNotificationEnabled = "false";
+            Registry configRegistry = ServiceReferenceHolder.getInstance().getRegistryService().
+                    getConfigSystemRegistry(tenantId);
+            if (configRegistry.resourceExists(APIConstants.API_TENANT_CONF_LOCATION)) {
+                Resource resource = configRegistry.get(APIConstants.API_TENANT_CONF_LOCATION);
+                String content = new String((byte[]) resource.getContent(), Charset.defaultCharset());
+                if (content != null) {
+                    JSONObject tenantConfig = (JSONObject) new JSONParser().parse(content);
+                    isNotificationEnabled = (String) tenantConfig.get(NotifierConstants.NOTIFICATIONS_ENABLED);
+                }
+            }
+
+            if (JavaUtils.isTrueExplicitly(isNotificationEnabled)) {
+
```

Comment:
Remove the unwanted blank line

Reply:
Fixed with 848e9c064209cd244b6df83544580342f97793be